required/tools/array.dtx

```diff
@@ -1388,11 +1389,29 @@ Bug reports can be opened (category \texttt{#1}) at\\%
 %    Here, we assume that the \textsf{count} register
 %    =\@tempcnta= has saved the value $=\count@= - 1$.
 %
-%    We end with \cs{relax} to stop any parsing for optional
-%    arguments out of \verb=>{...}= at this point. 
+%    To keep \TeX{} happy if there is a look ahead in the tabular preamble
+%    which uses the Appendix~D trick (for example anything with a trailing
+%    optional argument defined by \pkg{ltcmd}), we wrap everything here in
+%    a protected version of \cs{@firstofone}. \TeX{} otherwise can get
+%    confused about the value of the master counter, and we get some
+%    strange errors. (Quite possibly the underlying issue is a \TeX{}
```

Comment:
We might confess in a footnote that we don't quite understand what's happening right now ... maybe somebody sets us straight or finds the bug if there is one

Reply:
I've added something